### PR TITLE
Build and package war for new releases

### LINKED ISSUE #1947: Add Java WAR build task
We could add a `.war` (or `.jar`?) file to the release assets for new releases, so users can run gollum without having to install (J)Ruby. Context: https://github.com/gollum/gollum/issues/1892#issuecomment-1397022590

- [x] Create the build task
- [x] Document how to use gollum as a `.war` in the README and wiki

### PR DESCRIPTION
**Pull request by dometto**
Resolves #1947

I've tested this on my [personal fork](https://github.com/dometto/gollum/releases/tag/v17.5b).

### CODE REVIEW

**Comment by dometto** [comment]
If someone could sanity check the language in the readme that would be great :)

File: Gemfile [hunk 1]
```diff
@@ -1,6 +1,6 @@
 source 'https://rubygems.org'
 
-gem 'warbler', platforms: :jruby
+gem 'warbler', git: 'https://github.com/jruby/warbler' if RUBY_PLATFORM == 'java'
```
Comment (bartkamphorst) [inline Gemfile:3]: Why use this direct URL?

**Review by bartkamphorst — APPROVED** [review]
Great addition! Just one question about the Gemfile, otherwise it looks good to me.

File: Gemfile (hunk 1 above)
Comment (dometto) [reply Gemfile:3]: The latest gem release of warbler is sadly long ago, but master contains fixes we need for the build.

File: README.md [hunk 2]
```diff
@@ -37,7 +35,11 @@ Gollum is a simple wiki system built on top of Git. A Gollum Wiki is simply a gi
 
 ### SYSTEM REQUIREMENTS
 
-Gollum runs on Unix-like systems using its default [adapter](https://github.com/gollum/rugged_adapter) for [rugged](https://github.com/libgit2/rugged). You can also run Gollum on [JRuby](https://github.com/jruby/jruby) via its [adapter](https://github.com/repotag/gollum-lib_rjgit_adapter) for [RJGit](https://github.com/repotag/rjgit/). On Windows, Gollum runs only on JRuby.
+Gollum runs both on Unix-like systems and on Windows.
+
+Gollum runs either on 'normal' Ruby (MRI), or on Java via [JRuby](https://github.com/jruby/jruby). On Windows, Gollum runs only on JRuby/Java (either from source, or [prebuilt](#as-an-executable-Java-app])).
```
Comment (bartkamphorst) [inline README.md:40]: Maybe say that gollum runs on the JVM with JRuby rather than on Java.

File: README.md [hunk 3]
```diff
@@ -37,7 +35,11 @@ Gollum is a simple wiki system built on top of Git. A Gollum Wiki is simply a gi
...
+
+On MRI, Gollum uses the [rugged](https://github.com/libgit2/rugged) git library, while on Java it utilizes the [rjgit](https://github.com/repotag/rjgit), via its different [adapters](https://github.com/gollum/gollum/wiki/Git-adapters).
```
Comment (bartkamphorst) [inline README.md:42]: Same here, I would suggest replacing 'on Java' with 'on JRuby'.

File: README.md [hunk 4]
```diff
@@ -59,6 +61,12 @@ To run, simply:
 
 See [here](https://github.com/gollum/gollum/wiki/Gollum-via-Docker) for instructions on how to run Gollum via Docker.
 
+### As an executable Java app
+
+The [latest Release](https://github.com/gollum/gollum/releases/) of Gollum will always contains a downloadable `gollum.war` file that can be directly executed on any system with a working Java installation:
```
Comment (bartkamphorst) [inline README.md:66]: Typo. Will always `contain`.

File: README.md [hunk 5]
```diff
@@ -59,6 +61,12 @@ To run, simply:
 
 See [here](https://github.com/gollum/gollum/wiki/Gollum-via-Docker) for instructions on how to run Gollum via Docker.
 
+### As an executable Java app
```
Comment (bartkamphorst) [inline README.md:64]: Maybe call this heading `As a web application resource (requires Java)`?

**Review by bartkamphorst — CHANGES_REQUESTED** [review]
Couple of minor improvement suggestions in the README.

**Comment by dometto** [comment]
Thanks for the suggestions!

**Review by bartkamphorst — APPROVED** [review]
Thanks for making the changes, looks good!

**Pull request merged**